mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,801 @@
+                of background proposals in video duration.
+                Default: 0.01.
+            incomplete_overlap_threshold (float): Minimum percent of incomplete
+                proposals' own span contained in a groundtruth instance.
+                Default: 0.7.
+
+        Returns:
+            list[:obj:`SSNInstance`]: (incompletes, backgrounds), incompletes
+                and backgrounds are lists comprised of incomplete
+                proposal instances and background proposal instances.
+        """
+        incompletes = []
+        backgrounds = []
+
+        for proposal in proposals:
+            if (proposal.best_iou < incomplete_iou_threshold
+                    and proposal.overlap_self > incomplete_overlap_threshold):
+                incompletes.append(proposal)
+            elif (proposal.best_iou < background_iou_threshold
+                  and proposal.coverage > background_coverage_threshold):
+                backgrounds.append(proposal)
+
+        return incompletes, backgrounds
+
+    def _video_centric_sampling(self, record):
+        """Sample proposals from the this video instance.
+
+        Args:
+            record (dict): In ion of the video instance(video_info[idx]).
+                key: frame_dir, video_id, total_frames,
+                     gts: List of groundtruth instances(:obj:`SSNInstance`).
+                     proposals: List of proposal instances(:obj:`SSNInstance`).
+        """
+        positives = self.get_positives(record['gts'], record['proposals'],
+                                       self.assigner.positive_iou_threshold,
+                                       self.sampler.add_gt_as_proposals)
+        incompletes, backgrounds = self.get_negatives(
+            record['proposals'], self.assigner.incomplete_iou_threshold,
+            self.assigner.background_iou_threshold,
+            self.assigner.background_coverage_threshold,
+            self.assigner.incomplete_overlap_threshold)
+
+        def sample_video_proposals(proposal_type, video_id, video_pool,
+                                   num_requested_proposals, dataset_pool):
+            """This method will sample proposals from the this video pool. If
+            the video pool is empty, it will fetch from the dataset pool
+            (collect proposal of the entire dataset).
+
+            Args:
+                proposal_type (int): Type id of proposal.
+                    Positive/Foreground: 0
+                    Negative:
+                        Incomplete: 1
+                        Background: 2
+                video_id (str): Name of the video.
+                video_pool (list): Pool comprised of proposals in this video.
+                num_requested_proposals (int): Number of proposals
+                    to be sampled.
+                dataset_pool (list): Proposals of the entire dataset.
+
+            Returns:
+                list[(video_id, :obj:`SSNInstance`), proposal_type]
```

Comment:
This return docstring can not be automatically transferred to docs.

may be ``list[(int, :obj:`SSNInstance`), str]`` ? Need to double check